ember/app/services/filtered-forms.js

```diff
@@ -22,43 +23,26 @@ export default class FilteredFormsService extends Service {
       })
     ).filterBy("isAuthorized");
 
-    const isExpertAssociation = organisations.isAny("isExpertAssociation");
-    const isAdvisoryBoard = organisations.isAny("isAdvisoryBoard");
-
-    const allForms = {
-      ...(await this.apollo.query({
-        query: getFilteredFormsQuery,
-      })),
+    const userVisibilities = {
+      internalForm: this.abilities.can("create internal form case"),
+      advisoryBoardForm: organisations.isAny("isAdvisoryBoard"),
+      expertAssociationForm: organisations.isAny("isExpertAssociation"),
     };
+    const allForms = await this.mainForms();
 
-    if (!isExpertAssociation) {
-      delete allForms.expertAssociation;
-    }
-    if (!isAdvisoryBoard) {
-      delete allForms.advisoryBoard;
-    }
-    if (!isExpertAssociation && !isAdvisoryBoard) {
-      delete allForms.bothTypes;
-    }
-
-    const value = [
-      allForms?.expertAssociation,
-      allForms?.advisoryBoard,
-      allForms?.bothTypes,
-      allForms?.public,
-    ]
-      .map((forms) => {
-        return forms?.edges.map((edge) => edge.node) ?? [];
-      })
-      .flat();
+    this.value = allForms.filter((form) => {
+      const visibilities = ENV.APP.caluma.formVisibilities.map((visibility) => {
+        return form.meta[visibility] && userVisibilities[visibility];
+      });
+      const specificVisibility = visibilities.any((visibility) => visibility);
+      const publicVisibility = visibilities.every(
```

Comment:
Maybe we could short circuit this here since if every visibility is undefined, the code above does not need to be executed.

Reply:
done